samples/multicluster/setup-mesh.sh

```diff
@@ -0,0 +1,326 @@
+
+  kc apply -f - <<EOF
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: cluster-aware-gateway
+  namespace: istio-system
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+  - port:
+      number: 443
+      name: tls
+      protocol: TLS
+    tls:
+      mode: AUTO_PASSTHROUGH
+    hosts:
+    - "*.local"
+EOF
+}
+
+mesh_contexts() {
+  sed -n 's/^  \([^ ]\+\):$/\1/p' "${MESH_TOPOLOGY_FILENAME}" | tr '\n' ' '
+}
+
+# apply the desired configuration to create the multi-cluster mesh.
+apply() {
+  echo "applying configuration to build the mesh"
+
+  # Step 1) Install the intermediate certificate and Istio control
+  # plane in each cluster
+  for CONTEXT in $(mesh_contexts); do
+      apply_intermediate_ca "${CONTEXT}"
+      apply_istio_control_plane "${CONTEXT}"
+  done
+
+  # Step 2) The ingress gateway's external IP is required for
+  # cross-network traffic. Wait for the external IPs to be
+  # allocated before proceeding.
+  for CONTEXT in $(mesh_contexts); do
+    wait_for_gateway "${CONTEXT}"
+  done
+
+  # Step 3) re-generate and apply configuration. The updated
+  # configuration includes the allocated external IPs from step (2).
+  for CONTEXT in $(mesh_contexts); do
+      apply_istio_control_plane "${CONTEXT}"
+
+      # required for multi-network mesh.
+      create_passthrough_gateway "${CONTEXT}"
+  done
+
+  # Step 4) Synchronize each cluster's service registry with the
+  # other clusters in the mesh. This is required for cross-cluster
+  # load balancing.
+  istioctl x multicluster apply -f "${MESH_TOPOLOGY_FILENAME}"
+}
+
+# Teardown the mesh. Delete all of the in-cluster resources that
+# may have been installed during setup.
+teardown() {
+  echo "Tearing down the mesh"
+
+  for CONTEXT in $(mesh_contexts); do
+    kc() { kubectl --context "${CONTEXT}" "$@"; }
+
+    echo "Tearing down control cluster ${CONTEXT}"
+
+    # wait for galley to die so it doesn't re-create the webhook config.
+    kc delete namespace istio-system  --wait --ignore-not-found || true
+    kc delete mutatingwebhookconfiguration -l operator.istio.io/managed=Reconcile --ignore-not-found
+    kc delete validatingwebhookconfiguration -l operator.istio.io/managed=Reconcile --ignore-not-found
+
+    # remove cluster scoped resources
+    kc delete clusterrole -l operator.istio.io/managed=Reconcile --ignore-not-found
+    kc delete clusterrolebinding -l operator.istio.io/managed=Reconcile --ignore-not-found
+    kc delete crd -l operator.istio.io/managed=Reconcile --ignore-not-found
+  done
+}
+
+describe() {
+  istioctl x mc describe -f "${MESH_TOPOLOGY_FILENAME}"
+}
+
+check_prerequisties() {
+  if [[ ! -d "${WORKDIR}" ]]; then
+    echo "error: workspace directory '${WORKDIR}' not found. Run 'mkdir workspace && export WORKDIR=\$(pwd)/workspace' "
+    exit 1
+  fi
+}
+
+usage() {
+  echo "Usage: $0 create-mesh | add-cluster | remove-cluster | apply | teardown
+
+create-mesh
```

Comment:
Should this be called something else since it's not really creating a mesh? How about just "prep"?

Reply:
done